src/main/java/org/sagebionetworks/bridge/services/AdherenceService.java

```diff
@@ -410,7 +419,14 @@ public WeeklyAdherenceReport getWeeklyAdherenceReport(String appId, String study
         report.setClientTimeZone(zoneId);
         
         WeeklyAdherenceReport weeklyReport = deriveWeeklyAdherenceFromStudyReportWeek(studyId, account, report);
-        
+
+        // trigger alert for low weekly adherence
+        if (weeklyReport.getWeeklyAdherencePercent() != null
+                && weeklyReport.getWeeklyAdherencePercent() <= WEEKLY_ADHERENCE_PERCENT_ALERT_THRESHOLD) {
+            alertService.createAlert(
+                    Alert.lowAdherence(studyId, appId, account.getId(), WEEKLY_ADHERENCE_PERCENT_ALERT_THRESHOLD));
+        }
```

Comment:
As per discussion, there is already a hook in the Adherence Report Worker that calculates adherence and checks against the adherence threshold. Do we want to use that?

Also, in the worker code, it uses less than, not less than or equal to. Can we confirm requirements?

Reply:
I'll check on that

Reply:
Confirmed that less than or equal to is correct